cmd/registry/controller/parser.go

```diff
@@ -42,13 +42,15 @@ func extendDependencyPattern(
 		return fmt.Sprintf("projects/%s/locations/global/%s", projectID, dependencyPattern), nil
 	}
 
-	entityRegex := regexp.MustCompile(fmt.Sprintf(`\%s\.(api|version|spec|artifact)`, resourceKW))
+	entityRegex := regexp.MustCompile(fmt.Sprintf(`(\%s\.(api|version|spec|artifact))(/|$)`, resourceKW))
 	matches := entityRegex.FindStringSubmatch(dependencyPattern)
-	if len(matches) <= 1 {
-		return "", errors.New(fmt.Sprintf("Invalid source pattern: %s", dependencyPattern))
+	// dependencyPattern: "$resource.api/artifacts/score"
+	// matches: ["$resource.api/", "$resource.api", "api"]
+	if len(matches) <= 2 {
+		return "", errors.New(fmt.Sprintf("Invalid dependency pattern: %s", dependencyPattern))
```

Comment:
Nit: `errors.New(fmt.Sprintf(...))` is equivalent to `fmt.Errorf(...)`. Also, error messages (only in the `error` type) should not be capitalized since they're often nested within other errors.

Reply:
Done.